app/modules/course/views/CourseView.js

```diff
@@ -7,20 +7,63 @@ define(function(require) {
         template: _.template(require("text!../templates/courseTemplate.html")),
         el: false,
         events: {
-            'click .btn': "subscribeDialog"
+            'click .btn-success': "subscribeDialog",
+            'click .btn-unsubscribe': 'unSubscribeDialog',
         },
         initialize: function() {
             this.subscribeModal = new CMS.ModalView({model: this.model});
+            this.subscribeUrl = CMS.api + "courses/subscribe";
         },
         serialize: function() {
             var course = this.model;
             course.attributes.id = this.model.id;
             course.parseDate = this.convertToMonthAndDate;
             return {
-                course: course
+                course: course,
+                courses: CMS.SessionModel.getItem("UserSession").profile._courses
             };
         },
-        subscribeDialog: function(ev){
+        subscribeDialog: function(e){
+            var _this = this;
+            CMS.ModalView.prototype.submitHandlerClick = function() {
+                $.ajax({
+                    url: _this.subscribeUrl,
+                    method: "POST",
+                    data: {id: this.model.id, name: this.model.get("name")},
+                    beforeSend: function(jqXHR){
```

Comment:
1. Put ajax method to model. And use model method to send ajax.
2. `beforeSend` property put to `ajaxSetup` functionality